CromIAM/src/test/scala/cromiam/webservice/SubmissionSupportSpec.scala

```diff
@@ -0,0 +1,41 @@
+package cromiam.webservice
+
+import akka.event.{LoggingAdapter, NoLogging}
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.headers.{Authorization, OAuth2BearerToken, RawHeader}
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import cromwell.util.SampleWdl.HelloWorld
+import org.scalatest.{FlatSpec, Matchers}
+import spray.json.DefaultJsonProtocol._
+import spray.json._
+
+class SubmissionSupportSpec extends FlatSpec with Matchers with ScalatestRouteTest with SubmissionSupport {
+  override val cromwellClient = new MockCromwellClient()
+  override val samClient = new MockSamClient()
+  override val log: LoggingAdapter = NoLogging
+
+  val authorization = Authorization(OAuth2BearerToken("my-token"))
+  val badAuthHeaders: List[HttpHeader] = List(authorization, RawHeader("OIDC_CLAIM_user_id", samClient.unauthorizedUserCollectionStr))
+  val goodAuthHeaders: List[HttpHeader] = List(authorization, RawHeader("OIDC_CLAIM_user_id", samClient.authorizedUserCollectionStr))
+
+  val submitPath: String = "/api/workflows/v1"
+
+  val helloWorldWdl = HelloWorld.workflowSource()
+  val helloWorldInputs = HelloWorld.rawInputs
+
+  val workflowSource = Multipart.FormData.BodyPart("workflowSource", HttpEntity(MediaTypes.`application/json`, helloWorldWdl))
+  val workflowInputs = Multipart.FormData.BodyPart("workflowInputs", HttpEntity(MediaTypes.`application/json`, helloWorldInputs.toJson.toString))
+  val formData = Multipart.FormData(workflowSource, workflowInputs).toEntity()
+
+  "Submit query" should "forward the request to Cromwell for authorized SAM user" in {
+    Post(submitPath).withHeaders(goodAuthHeaders).withEntity(formData) ~> submitRoute ~> check {
+      status shouldEqual StatusCodes.InternalServerError
```

Comment:
I haven't dug into the mocks here but a 500 seems strange for a "success" expectation. 🙂